libraries/libfc/test/crypto/test_cypher_suites.cpp

```diff
@@ -60,7 +60,7 @@ BOOST_AUTO_TEST_CASE(test_r1_recovery) try {
    auto pub = key.get_public_key();
    auto sig = key.sign(digest);
 
-   auto recovered_pub = public_key(sig, digest);
+   auto recovered_pub = public_key(sig, digest, fc::check_canonical_t::yes);
```

Comment:
It seems like these might be the only cases of `fc::check_canonical_t::yes`?

Does it make sense to just remove this ability from the `public_key` ctor entirely? Because other then these two lines[^1] the approach seems to be "always not care" about canonical at recovery time, and then check `is_ecc_canonical()` as appropriate (based on protocol feature activation). And this approach is really nice and clean.

So these two usages of `fc::check_canonical_t::yes` and all the plumbing needed in the ctor to support it sticks out a little.

But I can understand not wanting to make more changes than necessary.

[^1]: Which can be replaced with a `sig.is_ecc_canonical(); public_key(... fc::check_canonical_t::no); `